edb/testbase/connection.py

```diff
@@ -283,22 +283,29 @@ async def _ensure_transaction(self):
                 "Only managed retriable transactions are supported. "
                 "Use `async with transaction:`"
             )
-        if not self.__started:
-            self.__started = True
+        if not self._started:
+            self._started = True
             if self._connection.is_closed():
                 await self._connection.connect(
                     single_attempt=self.__iteration != 0
                 )
             await self.start()
 
 
+class RawIteration(RawTransaction, Iteration):
+    async def _ensure_transaction(self):
+        self._started = True
+        await super()._ensure_transaction()
+
+
 class Retry:
-    def __init__(self, connection):
+    def __init__(self, connection, raw=False):
```

Comment:
Or should we make all our `Iteration`s be `RawTransaction`s?

Reply:
I ended up taking a different approach that allows Iteration to always contain nested transactions, but not to be in one